File: backend/app/v1/schemas/transcript.py
"""
Pydantic schemas for transcript upload and retrieval.
"""

from __future__ import annotations

import uuid
from datetime import datetime

from pydantic import BaseModel


class DialogueTurn(BaseModel):
    """A single speaker turn in the transcript."""

    speaker: str
    timestamp: str
    text: str


class TranscriptMetadata(BaseModel):
    """Extracted metadata from the transcript document."""

    candidate_name: str = ""
    interviewer_name: str = ""
    interview_date: str = ""
    duration: str = ""
    position: str = ""


class TranscriptUploadResponse(BaseModel):
    """Returned immediately after a .docx transcript is uploaded."""

    recording_id: uuid.UUID
    transcript_id: uuid.UUID
    interview_id: uuid.UUID
    status: str  # "processing"
    message: str


class TranscriptStatusResponse(BaseModel):
    """Full transcript result once processing is complete."""

    transcript_id: uuid.UUID
    recording_id: uuid.UUID
    interview_id: uuid.UUID
    status: str  # "processing" | "completed" | "failed"
    metadata: TranscriptMetadata | None = None
    dialogue_count: int = 0
    dialogues: list[DialogueTurn] = []
    clean_text: str | None = None
    generated_at: datetime | None = None
    error: str | None = None

File: frontend/src/apis/types/transcript.ts
export interface DialogueTurn {
  speaker: string;
  timestamp: string;
  text: string;
}

export interface TranscriptMetadata {
  candidate_name: string;
  interviewer_name: string;
  interview_date: string;
  duration: string;
  position: string;
}

export type TranscriptStatus = "uploaded" | "processing" | "completed" | "failed";

export interface TranscriptUploadResponse {
  recording_id: string;
  transcript_id: string;
  interview_id: string;
  status: string;
  message: string;
}

export interface TranscriptStatusResponse {
  transcript_id: string;
  recording_id: string;
  interview_id: string;
  status: TranscriptStatus;
  metadata: TranscriptMetadata | null;
  dialogue_count: number;
  dialogues: DialogueTurn[];
  clean_text: string | null;
  generated_at: string | null;
  error: string | null;
}
